src/d3d9/d3d9_device.cpp

```diff
@@ -3783,10 +3783,14 @@ namespace dxvk {
       const bool newDepth = newTexture->IsShadow();
 
       if (oldDepth != newDepth) {
-        m_depthTextures &= ~(1u << StateSampler);
-        if (newDepth)
-          m_depthTextures |= 1u << StateSampler;
+        m_depthTextures ^= 1u << StateSampler;
```

Comment:
Can we separate out xor change into another commit in case we mess something up here?

Reply:
done